frontend/model/state.js

```diff
@@ -329,6 +335,235 @@ const getters = {
+        const existingPayment = {}
+        if (partialAmount > 0) {
+          // TODO/BUG this only work if the payment is done in 2 parts. if done in >=3 won't work.
+          const sentPartial = sent.find((s) => s.username === p.to && s.amount === partialAmount)
+          if (sentPartial) {
+            existingPayment.hash = sentPartial.hash // Q: Why do we need this hash?
+          }
+        }
+
+        payments.push({
+          ...existingPayment,
+          ...p,
+          total: existingAmount,
+          partial: partialAmount > 0,
+          dueIn
+        })
+      }
+      return payments
+    })()
+    const late = (() => {
+      // const currentDistribution = this.groupIncomeAdjustedDistribution
+      // const { pledgeAmount } = this.ourGroupProfile
+      const pMonthstamp = prevMonthstamp(cMonthstamp)
+      const latePayments = []
+      const pastMonth = monthlyPayments[pMonthstamp]
+      // QUESTION: Is this only from the previous month? What if there are payments from 2-3 months ago?
+      if (pastMonth) {
+        const pDate = dateFromMonthstamp(pMonthstamp)
+        const dueIn = lastDayOfMonth(pDate)
+        const adjusted = getters.groupIncomeAdjustedDistributionForMonth(pMonthstamp)
+
+        // This logic below is wrong (based on cypress tests). Commented for later analysis.
+        /*
+        for (const payment of pastMonth.lastAdjustedDistribution) {
+          if (payment.from === this.ourUsername && payment.amount > 0) {
+            // Let A = the amount we owe from the previous distribution.
+            // Let B = the total we've sent to payment.to from the current
+            //         month's paymentsFrom.
+            // Let C = the total amount we "owe" to payment.to from the
+            //         current month's distribution.
+            // Let D = the amount we're pledging this month
+            // Let E = the amount still unpaid for the previous month's distribution,
+            //         calculated as: C > 0 ? A : A + D - B
+            //
+            // If E > 0, then display a row for the late payment.
+            const A = payment.amount
+            const B = this.paymentTotalFromUserToUser(this.ourUsername, payment.to, cMonthstamp)
+            var C = currentDistribution
+              .filter(a => a.from === payment.from && a.to === payment.to)
+            C = C.length > 0 ? C[0].amount : 0
+            const D = pledgeAmount
+            const E = C > 0 ? A : A + D - B
+            if (E > 0) {
+              latePayments.push({
+                username: payment.to,
+                displayName: this.userDisplayName(payment.to),
+                amount: payment.amount, // TODO: include currency (what if it was changed?)
+                // partiaL: TODO add this info as it is in this.paymentsTodo
+                isLate: true,
+                date: dueIn
+              })
+            }
+          }
+        }
+        */
```

Comment:
>       // QUESTION: Is this only from the previous month? What if there are payments from 2-3 months ago?

We only support late payments for the previous month. We ignore all other months because it would be too complicated to deal with. We could get away without supporting late payments at all, but we have support for them, just for one month.

> // This logic below is wrong (based on cypress tests). Commented for later analysis.

That logic needs to be uncommented. If the tests aren't working then the logic needs to be fixed, not commented out.